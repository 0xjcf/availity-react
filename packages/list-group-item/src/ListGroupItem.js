import React from 'react';
import PropTypes from 'prop-types';
import { ListGroupItem as RsListGroupItem } from 'reactstrap';

const ListGroupItem = ({ borderColor, className, ...props }) => {
  const classname = [className, borderColor].filter(Boolean).join(' ');
  return <RsListGroupItem data-testid="list-group-item-id" {...props} className={classname} />;
};

ListGroupItem.propTypes = {
  /** The border color to display on the left of the item. Uses Availity UI Kit variants. Must be used within a `ListGroup` from `@availity/list-group` with the `cards` prop set to `true` */
  borderColor: PropTypes.string,
  /** The name of the class. */
  className: PropTypes.string,
};

export default ListGroupItem;
