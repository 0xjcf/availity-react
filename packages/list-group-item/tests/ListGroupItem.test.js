import React from 'react';
import renderer from 'react-test-renderer';
import ListGroupItem from '..';

describe('ListGroupItem', () => {
  test('should render', () => {
    const component = renderer.create(<ListGroupItem />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render color', () => {
    const component = renderer.create(<ListGroupItem color="success" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render border color', () => {
    const component = renderer.create(<ListGroupItem borderColor="success" />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

  test('should render border color and color', () => {
    const component = renderer.create(
      <ListGroupItem borderColor="success" color="success" />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
