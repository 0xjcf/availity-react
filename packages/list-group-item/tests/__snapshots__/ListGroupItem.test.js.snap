// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroupItem should render 1`] = `
<li
  className="list-group-item"
/>
`;

exports[`ListGroupItem should render border color 1`] = `
<li
  className="success list-group-item"
/>
`;

exports[`ListGroupItem should render border color and color 1`] = `
<li
  className="success list-group-item-success list-group-item"
/>
`;

exports[`ListGroupItem should render color 1`] = `
<li
  className="list-group-item-success list-group-item"
/>
`;
